frontend/src/metabase/components/ListItem/ListItem.jsx

```diff
@@ -1,42 +1,69 @@
 /* eslint "react/prop-types": "warn" */
-import { memo } from "react";
-import PropTypes from "prop-types";
 import cx from "classnames";
+import PropTypes from "prop-types";
+import { memo } from "react";
+
 import Card from "metabase/components/Card";
 import S from "metabase/components/List/List.css";
 import { Icon } from "metabase/core/components/Icon";
-import { ListItemLink, ListItemName } from "./ListItem.styled";
+import { ListItemLink, ListItemName, Root } from "./ListItem.styled";
 
-const ListItem = ({ name, description, placeholder, url, icon }) => (
-  <li className="relative">
-    <ListItemLink to={url}>
-      <Card hoverable className="mb2 p3 bg-white rounded bordered">
-        <div className={cx(S.item)}>
-          <div className={S.itemIcons}>
-            {icon && <Icon className={S.chartIcon} name={icon} size={16} />}
+const ListItem = ({
+  "data-testid": dataTestId,
+  name,
+  url,
+  description,
+  disabled,
+  placeholder,
+  icon,
+}) => {
+  const card = (
+    <Card hoverable className="mb2 p3 bg-white rounded bordered">
+      <div className={cx(S.item)}>
+        <div className={S.itemIcons}>
+          {icon && <Icon className={S.chartIcon} name={icon} size={16} />}
+        </div>
+        <div className={S.itemBody}>
+          <div className={S.itemTitle}>
+            <ListItemName
+              className={S.itemName}
+              tooltip={name}
+              tooltipMaxWidth="100%"
+            >
+              <h3>{name}</h3>
+            </ListItemName>
           </div>
-          <div className={S.itemBody}>
-            <div className={S.itemTitle}>
-              <ListItemName tooltip={name} tooltipMaxWidth="100%">
-                <h3>{name}</h3>
-              </ListItemName>
+          {(description || placeholder) && (
+            <div className={cx(S.itemSubtitle)}>
+              {description || placeholder}
             </div>
-            {(description || placeholder) && (
-              <div className={cx(S.itemSubtitle)}>
-                {description || placeholder}
-              </div>
-            )}
-          </div>
+          )}
         </div>
-      </Card>
-    </ListItemLink>
-  </li>
-);
+      </div>
+    </Card>
+  );
+
+  if (disabled) {
+    return (
+      <Root data-testid={dataTestId} disabled>
+        {card}
+      </Root>
+    );
+  }
+
+  return (
+    <Root data-testid={dataTestId}>
+      <ListItemLink to={url}>{card}</ListItemLink>
+    </Root>
+  );
```

Comment:
There are two differences in how this component works now:
- addition of `disabled` prop to conditionally render it without the link
- addition of `data-testid` prop